boot/seal.go

```diff
@@ -261,14 +261,20 @@ func sealKeyToModeenvForMethod(
 
 var resealKeyToModeenv = resealKeyToModeenvImpl
 
+type ResealKeyToModeenvOptions struct {
+	ExpectReseal      bool
+	IgnoreCache       bool
+	EnsureProvisioned bool
```

Comment:
these maybe should have doc comments, probably we also want to clarify that EnsureProvisioned doesn't mean reprovision